### PR TITLE
[Merged by Bors] - Support query model status

### PR DESCRIPTION
**Pull request by wangzhenyu32**
Colses #143 

Signed-off-by: wangzhenyu32 <wang.zhenyu32@zte.com.cn>

### CODE REVIEW

File: adlik_serving/apis/model_operate_impl.cc [hunk 1]
```diff
@@ -49,6 +49,59 @@ struct VersionDirectoryState : cub::DirentVisitor {
...
   VersionList versions;
 };
+
+struct QueryModelVerison : cub::DirentVisitor {
+  QueryModelVerison(ModelOperateResponse& rsp, std::string& modelName) : rsp(rsp), modelName(modelName) {
+  }
+
+private:
+  OVERRIDE(void visit(const std::string&, const std::string& child)) {
+    if (child != "config.pbtxt") {
```
Comment (shiyingjin) [inline adlik_serving/apis/model_operate_impl.cc:59]: It should be judged that child is the version directory.

File: adlik_serving/apis/model_operate.proto [hunk 2]
```diff
@@ -13,6 +13,16 @@ message ModelOperateRequest{
...
   string status = 1; //SUCCESS, ERROR
   string error_message = 2;
+  map<string, ModelInfo> models = 3;
 }
 
 
+
+message ModelInfo{
+  message Version{
+     string version_num = 1;
```
[the remark this answers is not in the document]
Comment (shiyingjin) [reply adlik_serving/apis/model_operate.proto:23]: "version_num" should be "int" better

File: adlik_serving/apis/model_operate_impl.cc [hunk 3]
```diff
@@ -49,6 +49,61 @@ struct VersionDirectoryState : cub::DirentVisitor {
...
 };
+
+struct QueryModelVerison : cub::DirentVisitor {
+  QueryModelVerison(ModelOperateResponse& rsp, std::string& modelName) : rsp(rsp), modelName(modelName) {
+  }
+
+private:
+  OVERRIDE(void visit(const std::string& base, const std::string& child)) {
+    uint32_t version_no = 1;
+    if (cub::strutils::to_uint32(child, version_no)) {
```
Comment (shiyingjin) [inline adlik_serving/apis/model_operate_impl.cc:60]: It should be judges that child is a directory

File: adlik_serving/apis/model_operate_impl.cc [hunk 4]
```diff
@@ -49,6 +49,61 @@ struct VersionDirectoryState : cub::DirentVisitor {
...
+struct QueryModelState : ManagedModelVisitor {
+  QueryModelState(ModelOperateResponse& rsp) : rsp(rsp) {
+  }
+
+private:
+  void visit(const Model& model) override {
+    auto versions = (*rsp.mutable_models())[model.getName()].mutable_versions();
+    for (auto iter = versions->begin(); iter != versions->end(); iter++) {
+      if (model.getVersion() == iter->version_no()) {
+        iter->set_state(model.str());
```
Comment (shiyingjin) [inline adlik_serving/apis/model_operate_impl.cc:99]: Should break here

File: cub/env/fs/file_system.h [hunk 5]
```diff
@@ -39,6 +39,7 @@ DEFINE_ROLE(FileSystem) {
   ABSTRACT(Status copyDir(const std::string& from, const std::string& to) const);
   ABSTRACT(Status deleteDir(const std::string& path) const);
   ABSTRACT(bool exists(const std::string&) const);
+  ABSTRACT(bool isDir(const std::string&) const);
```
Comment (shiyingjin) [inline cub/env/fs/file_system.h:42]: windows/env.cc hasn't implement this function

**Comment by amadeus-zte** [comment]
bors retry

**Pull request closed without merging**